packages/core/admin/server/validation/api-tokens.js

```diff
@@ -22,7 +27,12 @@ const apiTokenUpdateSchema = yup
     description: yup.string().nullable(),
     type: yup.string().oneOf(Object.values(constants.API_TOKEN_TYPE)).notNull(),
     permissions: yup.array().of(yup.string()).nullable(),
-    lifespan: yup.number().integer().min(1).nullable(),
+    lifespan: yup
+      .number()
+      .integer()
+      .min(1)
```

Comment:
Same here

Reply:
This was intentional, since it doesn't interfere and it's actually a requirement on its own even outside of the accepted values. That is, if we added a negative number to lifespans for some reason (like if it's replaced with a user-configurable field, typo/bug, etc), it should still fail because a negative will break regardless.

I'm curious about yup validation ordering now. Do you know if it throw all the errors, or just the first? If just the first, this should at least be moved after the oneOf since that error is more precise.

I'm not opposed to removing it if you feel strongly about it though :)

Reply:
Checked and yup should be validating everything in parallel, so all errors would be returned. So to me it makes sense to leave as an extra check.